src/backend/core/src/main/scala/com/lightbend/statefulserverless/Serve.scala

```diff
@@ -179,3 +176,114 @@ object Serve {
+        val responseCodec = Codecs.negotiate(req)
+        GrpcMarshalling.unmarshalStream(req)(ServerReflectionRequestSerializer, mat)
+        .map( _.alsoTo(Sink.foreach(x => println(x.toString))).via(handler))
+        .map(e => GrpcMarshalling.marshalStream(e, GrpcExceptionHandler.defaultMapper)(ServerReflectionResponseSerializer, mat, responseCodec, sys))
+    }
+  }
+
+  private final def findFileDescForName(name: String, fileDesc: FileDescriptor): Option[FileDescriptor] =
+    if (name == fileDesc.getName) Option(fileDesc)
+    else fileDesc.getDependencies.iterator.asScala.map(fd => findFileDescForName(name, fd)).find(_.isDefined).flatten
+
+  private final def containsSymbol(symbol: String, fileDesc: FileDescriptor): Boolean =
+    (symbol.startsWith(fileDesc.getPackage)) && // Ensure package match first
+    (Names.splitNext(if (fileDesc.getPackage.isEmpty) symbol else symbol.drop(fileDesc.getPackage.length + 1)) match {
+      case ("", "") => false
+      case (typeOrService, "") => 
+        fileDesc.findMessageTypeByName(typeOrService) != null ||
+        fileDesc.findServiceByName(typeOrService) != null
+      case (service, method) =>
+        Option(fileDesc.findServiceByName(service)).exists(_.findMethodByName(method) != null)
+    })
+
+  private final def findFileDescForSymbol(symbol: String, fileDesc: FileDescriptor): Option[FileDescriptor] =
+    if (containsSymbol(symbol, fileDesc)) Option(fileDesc)
+    else fileDesc.getDependencies.iterator.asScala.map(fd => findFileDescForSymbol(symbol, fd)).find(_.isDefined).flatten
+
+  private final def containsExtension(container: String, number: Int, fileDesc: FileDescriptor): Boolean =
+    fileDesc.getExtensions.iterator.asScala.exists(ext => container == ext.getContainingType.getFullName && number == ext.getNumber)
+
+  private final def findFileDescForExtension(container: String, number: Int, fileDesc: FileDescriptor): Option[FileDescriptor] =
+    if (containsExtension(container, number, fileDesc)) Option(fileDesc)
+    else fileDesc.getDependencies.iterator.asScala.map(fd => findFileDescForExtension(container, number, fd)).find(_.isDefined).flatten
+
+  private final def findExtensionNumbersForContainingType(container: String, fileDesc: FileDescriptor): List[Int] = 
+    fileDesc.getDependencies.iterator.asScala.foldLeft(
+      fileDesc.getExtensions.iterator.asScala.collect({ case ext if ext.getFullName == container => ext.getNumber }).toList
+    )((list, fd) => findExtensionNumbersForContainingType(container, fd) ::: list)
+
+  // TODO create caches for all of these lookups when/if needed
+  private def handle(fileDesc: FileDescriptor): Flow[ServerReflectionRequest, ServerReflectionResponse, NotUsed] =
+    Flow[ServerReflectionRequest].map { req =>
+      import ServerReflectionRequest.{ MessageRequest => In}
+      import ServerReflectionResponse.{ MessageResponse => Out}
+
+      val response = req.messageRequest match {
```

Comment:
Looks useful, might indeed be interesting to make this easy to use with akka-grpc in some form.

Reply:
:+1: